test/index.test.js

```diff
@@ -5,80 +5,134 @@ const co = require('co');
 const mongoose = require('mongoose');
 const mongooseLeanVirtuals = require('../');
 
-describe('tests', function() {
-  it('works', function(done) {
-    const schema = new mongoose.Schema({
-      name: String
-    });
-    schema.virtual('lowercaseName').get(function() {
-      return this.name.toLowerCase();
-    });
-    schema.plugin(mongooseLeanVirtuals);
+let baseDocId;
+let baseModel;
+let baseObj;
+let baseSchema;
+before(async () => {
+  mongoose.Promise = global.Promise;
+  await mongoose.connect('mongodb://localhost:27017/mongooseLeanVirtuals', {
```

Comment:
IMO, we should use [yield with co](https://www.npmjs.com/package/co) to function as `await` to maintain compatibility with node.js v4.

Reply:
@Fonger I agree :+1:

Reply:
Fair enough.  Would be good to explicitly state which versions of node you intend to support with this lib to help clarify this for future contributors.  Thanks for handling the change.